citrus-simulator 2.1.1 incompatible with citrus 3.4.0

Steps to reproduce:
- Use citrus-simulator 2.1.1 with citrus-ws 3.4.0 and invoke SOAP-request on a scenario

Expectation:
- Scenario runs successfuly

Result:
- Simulator returns a SOAP Fault: "java.lang.NullPointerException"

Stack Trace:
```
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:282)
	at org.citrusframework.simulator.service.ScenarioExecutionService.lambda$startScenarioAsync$3(ScenarioExecutionService.java:153)
	at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:367)
	at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:375)
	at org.citrusframework.simulator.service.ScenarioExecutionService.lambda$startScenarioAsync$5(ScenarioExecutionService.java:118)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.NoSuchMethodError: 'com.consol.citrus.actions.ReceiveMessageAction$ReceiveMessageActionBuilder com.consol.citrus.actions.ReceiveMessageAction$ReceiveMessageActionBuilder.endpoint(com.consol.citrus.endpoint.Endpoint)'
	at com.consol.citrus.dsl.builder.ReceiveMessageActionBuilder.endpoint(ReceiveMessageActionBuilder.java:78)
	at org.citrusframework.simulator.ws.SoapScenarioActionBuilder.receive(SoapScenarioActionBuilder.java:39)
	at org.citrusframework.simulator.ws.SoapScenarioRunnerActionBuilder.receive(SoapScenarioRunnerActionBuilder.java:34)
	...
```

Analysis:
It looks like scenario execution is no longer compatible after the refactoring in citrusframework/citrus#804. See https://github.com/citrusframework/citrus/commit/96612130cf5fbd46beead44e96edecafc791b2a9#diff-592163df49e4a545c79fdebfd5a342ec8f4bee537e78ddda15691db71ab1045fL522

--- a/simulator-archetypes/archetype-jms/src/main/resources/archetype-resources/src/main/resources/application.properties
+++ b/simulator-archetypes/archetype-jms/src/main/resources/archetype-resources/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=JMS Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable JMS support
 citrus.simulator.jms.enabled=true

--- a/simulator-archetypes/archetype-mail/src/main/resources/archetype-resources/src/main/resources/application.properties
+++ b/simulator-archetypes/archetype-mail/src/main/resources/archetype-resources/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=Mail Server Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable Mail endpoint support
 citrus.simulator.endpoint.enabled=true

--- a/simulator-archetypes/archetype-rest/src/main/resources/archetype-resources/src/main/resources/application.properties
+++ b/simulator-archetypes/archetype-rest/src/main/resources/archetype-resources/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=REST Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable Http REST support
 citrus.simulator.rest.enabled=true

--- a/simulator-archetypes/archetype-swagger/src/main/resources/archetype-resources/src/main/resources/application.properties
+++ b/simulator-archetypes/archetype-swagger/src/main/resources/archetype-resources/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=REST Petstore Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable Http REST support
 citrus.simulator.rest.enabled=true

--- a/simulator-archetypes/archetype-ws/src/main/resources/archetype-resources/src/main/resources/application.properties
+++ b/simulator-archetypes/archetype-ws/src/main/resources/archetype-resources/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=SOAP Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable SOAP web service support
 citrus.simulator.ws.enabled=true

--- a/simulator-archetypes/archetype-wsdl/src/main/resources/archetype-resources/src/main/resources/application.properties
+++ b/simulator-archetypes/archetype-wsdl/src/main/resources/archetype-resources/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=SOAP WSDL Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable SOAP web service support
 citrus.simulator.ws.enabled=true

--- a/simulator-docs/src/main/asciidoc/introduction.adoc
+++ b/simulator-docs/src/main/asciidoc/introduction.adoc
@@ -20,7 +20,6 @@ However, it is important to note that certain experimental features are included
 These features are introduced with "feature flags" and are still in the developmental phase:
 
 .Experimental Features
-
 *WSDL Generated Scenarios*: Automatically create simulator scenarios from a WSDL file.
 
 *Swagger Generated Scenarios*: Automatically create simulator scenarios from a Swagger API file.
@@ -55,3 +54,17 @@ This UI is best accessed through a web browser .
 It’s important to note that we do not perform cross-browser testing during our pipeline.
 Thus, we cannot guarantee compatibility in all browsers.
 Development and testing are primarily done on Chrome and Firefox, which are likely to yield the best experience.
+
+[[compatibility-matrix]]
+=== Compatibility Matrix
+
+The below table depicts the compatible version ranges of the https://github.com/citrusframework/citrus-simulator[`citrus-simulator`^], the https://github.com/citrusframework/citrus-simulator[citrusframework^] and https://spring.io/projects/spring-boot[Spring Boot^].
+
+The compatible java version is derived from the combination of the underlying frameworks.
+
+[cols="1,1,1,1"]
+|===
+| Simulator | Citrus | Spring-Boot | Java
+| `v3.0.0` | > `v4.2.0` | > `v3.2.5` | 17 to 21
+| `v2.1.1` | < `v3.4.0` | < `v3.0.0` | 8 to 11
+|===

--- a/simulator-docs/src/main/asciidoc/introduction.adoc
+++ b/simulator-docs/src/main/asciidoc/introduction.adoc
@@ -20,9 +20,10 @@ However, it is important to note that certain experimental features are included
 These features are introduced with "feature flags" and are still in the developmental phase:
 
 .Experimental Features
+
 *WSDL Generated Scenarios*: Automatically create simulator scenarios from a WSDL file.
+
 *Swagger Generated Scenarios*: Automatically create simulator scenarios from a Swagger API file.
-*Combined Simulator*: Operate multiple message transports within a single simulator instance.
 
 These features are evolving, and with the valuable insights we receive from our community, we are optimistic that they will achieve stability in due course.
 

--- a/simulator-samples/sample-bank-service/src/main/resources/application.properties
+++ b/simulator-samples/sample-bank-service/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=Citrus Bank Simulator
-logging.level.org.citrusframework=INFO
 
 # SSL Configuration for HTTPS endpoint
 server.port=8443

--- a/simulator-samples/sample-combined/src/main/resources/application.properties
+++ b/simulator-samples/sample-combined/src/main/resources/application.properties
@@ -2,7 +2,7 @@ info.simulator.name=Combined REST and JMS Simulator
 
 logging.level.org.citrusframework=INFO
 
-# Enable Http REST and JMS support
+# Enable HTTP REST and JMS and WS support
 citrus.simulator.rest.enabled=true
 citrus.simulator.jms.enabled=true
 citrus.simulator.ws.enabled=true

--- a/simulator-samples/sample-combined/src/main/resources/application.properties
+++ b/simulator-samples/sample-combined/src/main/resources/application.properties
@@ -1,7 +1,5 @@
 info.simulator.name=Combined REST and JMS Simulator
 
-logging.level.org.citrusframework=INFO
-
 # Enable HTTP REST and JMS and WS support
 citrus.simulator.rest.enabled=true
 citrus.simulator.jms.enabled=true

--- a/simulator-samples/sample-jms-fax/src/main/resources/application.properties
+++ b/simulator-samples/sample-jms-fax/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=JMS Async Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable JMS support
 citrus.simulator.jms.enabled=true

--- a/simulator-samples/sample-jms/src/main/resources/application.properties
+++ b/simulator-samples/sample-jms/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=JMS Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable JMS support
 citrus.simulator.jms.enabled=true

--- a/simulator-samples/sample-mail/src/main/resources/application.properties
+++ b/simulator-samples/sample-mail/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=Mail Server Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable Mail endpoint support
 citrus.simulator.endpoint.enabled=true

--- a/simulator-samples/sample-rest/src/main/java/org/citrusframework/simulator/sample/scenario/ParameterScenario.java
+++ b/simulator-samples/sample-rest/src/main/java/org/citrusframework/simulator/sample/scenario/ParameterScenario.java
@@ -1,0 +1,62 @@
+/*
+ * Copyright the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.citrusframework.simulator.sample.scenario;
+
+import org.citrusframework.http.message.HttpMessage;
+import org.citrusframework.simulator.scenario.AbstractSimulatorScenario;
+import org.citrusframework.simulator.scenario.Scenario;
+import org.citrusframework.simulator.scenario.ScenarioRunner;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.util.Map;
+
+import static org.citrusframework.actions.EchoAction.Builder.echo;
+
+@Scenario("Parameter")
+@RequestMapping(value = "/services/rest/simulator/parameter", method = RequestMethod.GET)
+public class ParameterScenario extends AbstractSimulatorScenario {
+
+    @Override
+    public void run(ScenarioRunner scenario) {
+        scenario.$(scenario.http()
+            .receive()
+            .get()
+            .getMessageBuilderSupport()
+            .extract((message, testContext) -> {
+                if (message instanceof HttpMessage httpMessage) {
+                    testContext.addVariables(Map.of("pulver", httpMessage.getQueryParams().get("exploded")));
+                }
+            }));
+
+        scenario.$(echo("Extracted dangerous parameters: ${pulver}"));
+
+        scenario.$(scenario.http()
+            .send()
+            .response(HttpStatus.OK)
+            .message()
+            .body(
+                // language=json
+                """
+                    {
+                        "pulver": ${pulver}
+                    }
+                    """
+            ));
+    }
+}

--- a/simulator-samples/sample-rest/src/main/resources/application.properties
+++ b/simulator-samples/sample-rest/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=REST Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable Http REST support
 citrus.simulator.rest.enabled=true

--- a/simulator-samples/sample-rest/src/test/java/org/citrusframework/simulator/SimulatorRestIT.java
+++ b/simulator-samples/sample-rest/src/test/java/org/citrusframework/simulator/SimulatorRestIT.java
@@ -49,7 +49,9 @@ public class SimulatorRestIT extends TestNGCitrusSpringSupport {
 
     private final String defaultResponse = "<DefaultResponse>This is a default response!</DefaultResponse>";
 
-    /** Test Http REST client */
+    /**
+     * Test Http REST client
+     */
     @Autowired
     private HttpClient simulatorClient;
 
@@ -59,21 +61,21 @@ public class SimulatorRestIT extends TestNGCitrusSpringSupport {
     @CitrusTest
     public void testHelloRequest() {
         $(http().client(simulatorClient)
-                .send()
-                .post("hello")
-                .message()
-                .contentType(MediaType.APPLICATION_XML_VALUE)
-                .body("<Hello xmlns=\"http://citrusframework.org/schemas/hello\">" +
-                            "Say Hello!" +
-                         "</Hello>"));
+            .send()
+            .post("hello")
+            .message()
+            .contentType(MediaType.APPLICATION_XML_VALUE)
+            .body("<Hello xmlns=\"http://citrusframework.org/schemas/hello\">" +
+                "Say Hello!" +
+                "</Hello>"));
 
         $(http().client(simulatorClient)
-                .receive()
-                .response(HttpStatus.OK)
-                .message()
-                .body("<HelloResponse xmlns=\"http://citrusframework.org/schemas/hello\">" +
-                            "Hi there!" +
-                         "</HelloResponse>"));
+            .receive()
+            .response(HttpStatus.OK)
+            .message()
+            .body("<HelloResponse xmlns=\"http://citrusframework.org/schemas/hello\">" +
+                "Hi there!" +
+                "</HelloResponse>"));
     }
 
     /**
@@ -82,23 +84,22 @@ public class SimulatorRestIT extends TestNGCitrusSpringSupport {
     @CitrusTest
     public void testHowdyRequest() {
         $(http().client(simulatorClient)
-                .send()
-                .post("howdy")
-                .message()
-                .contentType(MediaType.APPLICATION_XML_VALUE)
-                .body("<Hello xmlns=\"http://citrusframework.org/schemas/hello\">" +
-                        "Say Hello!" +
-                        "</Hello>"));
+            .send()
+            .post("howdy")
+            .message()
+            .contentType(MediaType.APPLICATION_XML_VALUE)
+            .body("<Hello xmlns=\"http://citrusframework.org/schemas/hello\">" +
+                "Say Hello!" +
+                "</Hello>"));
 
         $(http().client(simulatorClient)
-                .receive()
-                .response(HttpStatus.OK)
-                .message()
-                .body("<HelloResponse xmlns=\"http://citrusframework.org/schemas/hello\">" +
-                        "Howdy partner!" +
-                        "</HelloResponse>"));
+            .receive()
+            .response(HttpStatus.OK)
+            .message()
+            .body("<HelloResponse xmlns=\"http://citrusframework.org/schemas/hello\">" +
+                "Howdy partner!" +
+                "</HelloResponse>"));
     }
-
 
 
     /**
@@ -107,21 +108,21 @@ public class SimulatorRestIT extends TestNGCitrusSpringSupport {
     @CitrusTest
     public void testGoodByeRequest() {
         $(http().client(simulatorClient)
-                .send()
-                .post("goodbye")
-                .message()
-                .contentType(MediaType.APPLICATION_XML_VALUE)
-                .body("<GoodBye xmlns=\"http://citrusframework.org/schemas/hello\">" +
-                            "Say GoodBye!" +
-                         "</GoodBye>"));
+            .send()
+            .post("goodbye")
+            .message()
+            .contentType(MediaType.APPLICATION_XML_VALUE)
+            .body("<GoodBye xmlns=\"http://citrusframework.org/schemas/hello\">" +
+                "Say GoodBye!" +
+                "</GoodBye>"));
 
         $(http().client(simulatorClient)
-                .receive()
-                .response(HttpStatus.OK)
-                .message()
-                .body("<GoodByeResponse xmlns=\"http://citrusframework.org/schemas/hello\">" +
-                            "Bye bye!" +
-                         "</GoodByeResponse>"));
+            .receive()
+            .response(HttpStatus.OK)
+            .message()
+            .body("<GoodByeResponse xmlns=\"http://citrusframework.org/schemas/hello\">" +
+                "Bye bye!" +
+                "</GoodByeResponse>"));
     }
 
     /**
@@ -130,19 +131,19 @@ public class SimulatorRestIT extends TestNGCitrusSpringSupport {
     @CitrusTest
     public void testDefaultRequest() {
         $(http().client(simulatorClient)
-                .send()
-                .post()
-                .message()
-                .contentType(MediaType.APPLICATION_XML_VALUE)
-                .body("<Default>" +
-                            "Should trigger default scenario" +
-                        "</Default>"));
+            .send()
+            .post()
+            .message()
+            .contentType(MediaType.APPLICATION_XML_VALUE)
+            .body("<Default>" +
+                "Should trigger default scenario" +
+                "</Default>"));
 
         $(http().client(simulatorClient)
-                .receive()
-                .response(HttpStatus.OK)
-                .message()
-                .body(defaultResponse));
+            .receive()
+            .response(HttpStatus.OK)
+            .message()
+            .body(defaultResponse));
     }
 
     /**
@@ -153,57 +154,57 @@ public class SimulatorRestIT extends TestNGCitrusSpringSupport {
         variable("correlationId", "citrus:randomNumber(10)");
 
         $(http().client(simulatorClient)
-                .send()
-                .post("goodnight")
-                .message()
-                .contentType(MediaType.APPLICATION_XML_VALUE)
-                .body("<GoodNight xmlns=\"http://citrusframework.org/schemas/hello\">" +
-                            "Go to sleep!" +
-                        "</GoodNight>")
-                .header("x-correlationid", "${correlationId}"));
+            .send()
+            .post("goodnight")
+            .message()
+            .contentType(MediaType.APPLICATION_XML_VALUE)
+            .body("<GoodNight xmlns=\"http://citrusframework.org/schemas/hello\">" +
+                "Go to sleep!" +
+                "</GoodNight>")
+            .header("x-correlationid", "${correlationId}"));
 
         $(http().client(simulatorClient)
-                .receive()
-                .response(HttpStatus.OK)
-                .message()
-                .body("<GoodNightResponse xmlns=\"http://citrusframework.org/schemas/hello\">" +
-                            "Good Night!" +
-                        "</GoodNightResponse>"));
+            .receive()
+            .response(HttpStatus.OK)
+            .message()
+            .body("<GoodNightResponse xmlns=\"http://citrusframework.org/schemas/hello\">" +
+                "Good Night!" +
+                "</GoodNightResponse>"));
 
         //
         // Should be handled by default scenario
         //
 
         $(http().client(simulatorClient)
-                .send()
-                .post()
-                .message()
-                .contentType(MediaType.APPLICATION_XML_VALUE)
-                .body("<InterveningRequest>No match correlation!</InterveningRequest>"));
+            .send()
+            .post()
+            .message()
+            .contentType(MediaType.APPLICATION_XML_VALUE)
+            .body("<InterveningRequest>No match correlation!</InterveningRequest>"));
 
         $(http().client(simulatorClient)
-                .receive()
-                .response(HttpStatus.OK)
-                .message()
-                .body(defaultResponse));
+            .receive()
+            .response(HttpStatus.OK)
+            .message()
+            .body(defaultResponse));
 
         //
         // Should be handled by good-night scenario
         //
 
         $(http().client(simulatorClient)
-                .send()
-                .post()
-                .message()
-                .contentType(MediaType.APPLICATION_XML_VALUE)
-                .body("<InterveningRequest>In between!</InterveningRequest>")
-                .header("x-correlationid", "${correlationId}"));
+            .send()
+            .post()
+            .message()
+            .contentType(MediaType.APPLICATION_XML_VALUE)
+            .body("<InterveningRequest>In between!</InterveningRequest>")
+            .header("x-correlationid", "${correlationId}"));
 
         $(http().client(simulatorClient)
-                .receive()
-                .response(HttpStatus.OK)
-                .message()
-                .body("<InterveningResponse>In between!</InterveningResponse>"));
+            .receive()
+            .response(HttpStatus.OK)
+            .message()
+            .body("<InterveningResponse>In between!</InterveningResponse>"));
 
         $(sleep().milliseconds(2000L));
 
@@ -212,18 +213,39 @@ public class SimulatorRestIT extends TestNGCitrusSpringSupport {
         //
 
         $(http().client(simulatorClient)
-                .send()
-                .post()
-                .message()
-                .contentType(MediaType.APPLICATION_XML_VALUE)
-                .body("<InterveningRequest>After correlation!</InterveningRequest>")
-                .header("x-correlationid", "${correlationId}"));
+            .send()
+            .post()
+            .message()
+            .contentType(MediaType.APPLICATION_XML_VALUE)
+            .body("<InterveningRequest>After correlation!</InterveningRequest>")
+            .header("x-correlationid", "${correlationId}"));
 
         $(http().client(simulatorClient)
-                .receive()
-                .response(HttpStatus.OK)
-                .message()
-                .body(defaultResponse));
+            .receive()
+            .response(HttpStatus.OK)
+            .message()
+            .body(defaultResponse));
+    }
+
+    @CitrusTest
+    public void testSimulationWithExplodingQueryParams() {
+        $(http().client(simulatorClient)
+            .send()
+            .get("parameter")
+            .queryParam("exploded", "1")
+            .queryParam("exploded", "2"));
+
+        $(http().client(simulatorClient)
+            .receive()
+            .response(HttpStatus.OK)
+            .message()
+            .body(
+                """
+                    {
+                      "pulver": [1, 2]
+                    }
+                    """
+            ));
     }
 
     /**
@@ -293,8 +315,8 @@ public class SimulatorRestIT extends TestNGCitrusSpringSupport {
         @Bean
         public HttpClient simulatorClient() {
             return CitrusEndpoints.http().client()
-                    .requestUrl(String.format("http://localhost:%s/services/rest/simulator", 8080))
-                    .build();
+                .requestUrl(String.format("http://localhost:%s/services/rest/simulator", 8080))
+                .build();
         }
 
         @Bean

--- a/simulator-samples/sample-swagger/src/main/resources/application.properties
+++ b/simulator-samples/sample-swagger/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=REST Petstore Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable Http REST support
 citrus.simulator.rest.enabled=true

--- a/simulator-samples/sample-ws-client/src/main/resources/application.properties
+++ b/simulator-samples/sample-ws-client/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=SOAP Client Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable SOAP web service support
 citrus.simulator.ws.client.enabled=true

--- a/simulator-samples/sample-ws/src/main/resources/application.properties
+++ b/simulator-samples/sample-ws/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=SOAP Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable SOAP web service support
 citrus.simulator.ws.enabled=true

--- a/simulator-samples/sample-wsdl/src/main/resources/application.properties
+++ b/simulator-samples/sample-wsdl/src/main/resources/application.properties
@@ -1,5 +1,4 @@
 info.simulator.name=SOAP WSDL Simulator
-logging.level.org.citrusframework=INFO
 
 # Enable SOAP web service support
 citrus.simulator.ws.enabled=true

--- a/simulator-spring-boot/src/main/java/org/citrusframework/simulator/model/TestResult.java
+++ b/simulator-spring-boot/src/main/java/org/citrusframework/simulator/model/TestResult.java
@@ -39,7 +39,7 @@ import java.util.Arrays;
 import java.util.HashSet;
 import java.util.Set;
 
-import static jakarta.persistence.CascadeType.REMOVE;
+import static jakarta.persistence.CascadeType.ALL;
 import static java.util.Objects.nonNull;
 import static lombok.AccessLevel.NONE;
 import static org.apache.commons.lang3.exception.ExceptionUtils.getRootCause;
@@ -107,7 +107,7 @@ public class TestResult extends AbstractAuditingEntity<TestResult, Long> impleme
     /**
      * Optional test parameters
      */
-    @OneToMany(mappedBy = "testResult", cascade = {REMOVE})
+    @OneToMany(mappedBy = "testResult", cascade = ALL)
     @JsonIgnoreProperties(value = {"testResult"}, allowSetters = true)
     private final Set<TestParameter> testParameters = new HashSet<>();
 

--- a/simulator-spring-boot/src/test/java/org/citrusframework/simulator/service/impl/ScenarioExecutionServiceIT.java
+++ b/simulator-spring-boot/src/test/java/org/citrusframework/simulator/service/impl/ScenarioExecutionServiceIT.java
@@ -1,0 +1,58 @@
+package org.citrusframework.simulator.service.impl;
+
+import org.citrusframework.simulator.IntegrationTest;
+import org.citrusframework.simulator.model.ScenarioExecution;
+import org.citrusframework.simulator.model.TestParameter;
+import org.citrusframework.simulator.model.TestResult;
+import org.citrusframework.simulator.service.ScenarioExecutionService;
+import org.citrusframework.simulator.service.TestParameterService;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Isolated;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static java.time.Instant.now;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Isolated
+@IntegrationTest
+class ScenarioExecutionServiceIT {
+
+    @Autowired
+    private TestParameterService testParameterService;
+
+    @Autowired
+    private ScenarioExecutionService fixture;
+
+    @Nested
+    class CompleteScenarioExecution {
+
+        @Test
+        void cascadesUpdateEvent() {
+            var scenarioExecution = fixture.save(
+                ScenarioExecution.builder()
+                    .scenarioName("cascadesUpdateEvent")
+                    .startDate(now())
+                    .build()
+            );
+
+            var result = fixture.completeScenarioExecution(
+                scenarioExecution.getExecutionId(),
+                TestResult.builder()
+                    .testName("cascadesUpdateEvent")
+                    .className(getClass().getSimpleName())
+                    .build()
+                    .addTestParameter(
+                        TestParameter.builder()
+                            .key("key")
+                            .value("value")
+                            .build()));
+
+            assertThat(result.getTestResult().getTestParameters())
+                .allSatisfy(
+                    p -> assertThat(p.getTestParameterId().testResultId)
+                        .isNotNull()
+                );
+        }
+    }
+}

--- a/simulator-ui/src/main/java/org/citrusframework/simulator/ui/config/SimulatorUiConfigurationProperties.java
+++ b/simulator-ui/src/main/java/org/citrusframework/simulator/ui/config/SimulatorUiConfigurationProperties.java
@@ -43,7 +43,7 @@ public class SimulatorUiConfigurationProperties {
 
     public static class Security {
 
-        private static final String DEFAULT_CONTENT_SECURITY_POLICY = "default-src 'self'; frame-src 'self' data:;";
+        private static final String DEFAULT_CONTENT_SECURITY_POLICY = "default-src 'self'; frame-src 'self' data:; script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; font-src 'self' data:";
         private String contentSecurityPolicy = DEFAULT_CONTENT_SECURITY_POLICY;
 
         public String getContentSecurityPolicy() {

